Bottom: 35988cb00137c55b0074129532fb80e6fd9790d0
Top:    2c782e2c69cf659b160ae00328e5560543dca11a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-19 01:00:20 +0000

gerber.c: Tidy up use_gc() routine to avoid a static anlysis warning

Don't unconditionally access the aptr variable right after we
just NULL tested it.

(Caught by clang)


---

diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 5f5f349..3973814 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -951,7 +951,7 @@ use_gc (hidGC gc, int radius)
       if (aptr == NULL)
         pcb_fprintf (stderr, "error: aperture for width %$mS type %s is null\n",
                  linewidth, shape == ROUND ? "ROUND" : "SQUARE");
-      if (f)
+      else if (f)
 	fprintf (f, "G54D%d*", aptr->dCode);
     }
 }
